src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -0,0 +1,390 @@
+    }
+
+    public boolean getRepairPrimaryTokenRangeOnly(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.repair_primary_token_range_only);
+    }
+
+    public void setRepairPrimaryTokenRangeOnly(RepairType repairType, boolean primaryTokenRangeOnly)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).repair_primary_token_range_only = primaryTokenRangeOnly;
+    }
+
+    public int getParallelRepairPercentageInGroup(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.parallel_repair_percentage_in_group);
+    }
+
+    public void setParallelRepairPercentageInGroup(RepairType repairType, int percentageInGroup)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).parallel_repair_percentage_in_group = percentageInGroup;
+    }
+
+    public int getParallelRepairCountInGroup(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.parallel_repair_count_in_group);
+    }
+
+    public void setParallelRepairCountInGroup(RepairType repairType, int countInGroup)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).parallel_repair_count_in_group = countInGroup;
+    }
+
+    public boolean getMVRepairEnabled(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.mv_repair_enabled);
+    }
+
+    public void setMVRepairEnabled(RepairType repairType, boolean enabled)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).mv_repair_enabled = enabled;
+    }
+
+    public void setForceRepairNewNode(RepairType repairType, boolean forceRepairNewNode)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).force_repair_new_node = forceRepairNewNode;
+    }
+
+    public boolean getForceRepairNewNode(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.force_repair_new_node);
+    }
+
+    public String getTokenRangeSplitter(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.token_range_splitter);
+    }
+
+    // Options configures auto-repair behavior for a given repair type.
+    // All fields can be modified dynamically.
+    public static class Options implements Serializable
+    {
+        // The separator separating different DCs in repair_dc_groups
+        public static final String DC_GROUP_SEPARATOR = "\\|";
+
+        // defaultOptions defines the default auto-repair behavior when no overrides are defined
+        @VisibleForTesting
+        protected static final Options defaultOptions = getDefaultOptions();
+
+        public Options()
+        {
+        }
+
+        @VisibleForTesting
+        protected static Options getDefaultOptions()
+        {
+            Options opts = new Options();
+
+            opts.enabled = false;
```

Comment:
One thing worth discussing;  by default I noticed that if I set:

```yaml
auto_repair:
  enabled: true
```

That it doesn't actually run repairs because by default both full and incremental repair are disabled.  Is this how we want this to work?  I think it's a bit subjective and I don't have a good answer, but felt it was worth discussing.

I think two possible good defaults would be:

1. Keep it as-is, if you want to enable repair, you should have to choose which of the repair options you would want enabled by default.
2. Full repair is enabled with the above config.

I think maybe what we should just do is go with option one and add a commented section in cassandra.yaml that looks something like this:

```yaml
# Documentation about auto repair goes here
# auto_repair:
#  # Enables auto repair, defaults to false (disabled)
#  enabled: true
#  repair_type_overrides:
#    full:
#      # Enables auto full repairs, defaults to false
#      enabled: true
#     # ... all other various property defaults here
#    incremental:
#      enabled: false
```

Reply:
IMO, by default, we should just enable full repair. For the IR, we should let an operator decide whether they want to enable or not.

Reply:
that seems sensible to me :+1:.  I'm not sure whether or not this would get too tricky, but I could see that the 'defaults' for overrides could be contextual based on the kind of repair, e.g. I would want a more frequent `min_repair_interval_in_hours` for Incremental Repair than for Full Repair.    I think that may be too involved though, I think it's ok to just special case full repair to be enabled by default if you have `auto_repair.enabled: true`

Reply:
Done. I changed the `min_repair_interval_in_hours` to `DurationSpec.IntSecondsBound` so we can now go up to the second level of granularity.